src/main/java/eu/nimble/service/delegate/Delegate.java

```diff
@@ -401,8 +442,300 @@ public Response postPartySearch(Map<String, Object> body) {
+
+        return Response.status(Response.Status.OK)
+                .type(MediaType.APPLICATION_JSON)
+                .entity(result)
+                .build();
+    }
+
+    @GET
+    @Path("/catalogue/{catalogueUuid}/catalogueline/{lineId}/local")
+    public Response getCatalogueLine(@Context HttpHeaders headers, @PathParam("catalogueUuid") String catalogueUuid, @PathParam("lineId") String lineId) {
+        HashMap<String, List<String>> queryParams = new HashMap<String, List<String>>();
+        String path = String.format(getCatalogueLinePath,catalogueUuid,lineId);
+        path = catalogueServicePathPrefix != null ? catalogueServicePathPrefix + "/" + path : path;
+        URI uri = buildUri(catalogueServiceBaseUrl, catalogueServicePort, path, queryParams);
+        logger.info("got a request to endpoint " + String.format(getCatalogueLinePath,catalogueUuid,lineId) + ", forwarding to " + uri.toString());
+
+        return sendGetRequest(uri);
+    }
+
+    /***********************************   /catalogue/{catalogueUuid}/catalogueline/{lineId} - END *********************************/
+
+    /***********************************   /catalogue/{catalogueUuid}/cataloguelines ***************************************/
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/catalogue/{catalogueUuid}/cataloguelines")
+    public Response federatedGetCatalogueLines(@Context HttpHeaders headers, @PathParam("catalogueUuid") String catalogueUuid,@QueryParam("lineIds") List<String> lineIds, @QueryParam("delegateId") String delegateId) {
+        logger.info("called federated get catalogue lines");
+
+        String lineIdsQueryParam = "";
+        int size = lineIds.size();
+        for(int i = 0; i < size; i++){
+            if(i == size-1){
+                lineIdsQueryParam += lineIds.get(i);
+            }
+            else{
+                lineIdsQueryParam += lineIds.get(i) + ",";
+            }
+        }
+
+        HashMap<String, String> queryParams = new HashMap<String, String>();
+        queryParams.put("lineIds",lineIdsQueryParam);
+
+        String path = String.format(getCatalogueLinesLocalPath,catalogueUuid);
+        String result = sendGetRequest(path,queryParams,delegateId);
+
+        return Response.status(Response.Status.OK)
+                .type(MediaType.APPLICATION_JSON)
+                .entity(result)
+                .build();
+    }
+
+    @GET
+    @Path("/catalogue/{catalogueUuid}/cataloguelines/local")
+    public Response getCatalogueLines(@Context HttpHeaders headers, @PathParam("catalogueUuid") String catalogueUuid,@QueryParam("lineIds") List<String> lineIds) {
+
+        String lineIdsQueryParam = "";
+        int size = lineIds.size();
+        for(int i = 0; i < size; i++){
+            if(i == size-1){
+                lineIdsQueryParam += lineIds.get(i);
+            }
+            else{
+                lineIdsQueryParam += lineIds.get(i) + ",";
+            }
+        }
```

Comment:
lineIdsQueryParam implementation can be much simpler.
e.g:
for (String lineId : lineIds) {
        uriBuilder.queryParam("lineIds", lineId);
}